drivers/wifi/siwx91x/siwx91x_wifi_socket.c

```diff
@@ -84,30 +84,56 @@ void siwx91x_on_join_ipv4(struct siwx91x_dev *sidev)
 
 void siwx91x_on_join_ipv6(struct siwx91x_dev *sidev)
 {
+	if (!IS_ENABLED(CONFIG_NET_IPV6)) {
+		return;
+	}
+
 	sl_net_ip_configuration_t ip_config6 = {
 		.mode = SL_IP_MANAGEMENT_DHCP,
 		.type = SL_IPV6,
 	};
-	struct in6_addr addr6 = { };
-	int ret;
+	struct in6_addr addr6 = {0};
+	struct in6_addr ll_addr6 = {0};
+	struct in6_addr gw_addr6 = {0};
+	int ret = sl_si91x_configure_ip_address(&ip_config6, SL_SI91X_WIFI_CLIENT_VAP_ID);
 
-	if (!IS_ENABLED(CONFIG_NET_IPV6)) {
+	if (ret) {
+		LOG_ERR("sl_si91x_configure_ip_address(): %#04x", ret);
 		return;
 	}
-	/* FIXME: support for static IP configuration */
-	ret = sl_si91x_configure_ip_address(&ip_config6, SL_SI91X_WIFI_CLIENT_VAP_ID);
-	if (!ret) {
-		ARRAY_FOR_EACH(addr6.s6_addr32, i) {
-			addr6.s6_addr32[i] = ntohl(ip_config6.ip.v6.global_address.value[i]);
+
+	/* Extract IPv6 gateway address */
+	for (int i = 0; i < 4; i++) {
+		gw_addr6.s6_addr32[i] = ntohl(ip_config6.ip.v6.gateway.value[i]);
+	}
+	if (!IN6_IS_ADDR_UNSPECIFIED(&gw_addr6)) {
+		struct net_if_router *ret_gw =
+			net_if_ipv6_router_add(sidev->iface, &gw_addr6, 1800);
+
+		if (!ret_gw) {
+			LOG_ERR("Failed to add IPv6 gateway");
```

Comment:
`net_if_ipv6_addr_add()` also return a value. I suggest to either check all or none.